src/tensor_product_layer.jl

```diff
@@ -9,7 +9,7 @@ struct TensorLayer{M<:Array{TensorProductBasis},P<:AbstractArray,Int} <: Abstrac
         for basis in model
             number_of_weights *= basis.n
         end
-        p = rand(out*number_of_weights)
+        p = zeros(out*number_of_weights)
```

Comment:
```suggestion
        p = randn(out*number_of_weights)
```

Having them all positive is hard to fit, having them all negative is hard to fit, having them all zero makes it very poorly conditioned, so normally Wishart or normally distributed random numbers are effective.